addons/mail/models/mail_message.py

```diff
@@ -948,6 +948,17 @@ def _message_format(self, fnames):
         self.check_access_rule('read')
         vals_list = self._read_format(fnames)
         safari = request and request.httprequest.user_agent.browser == 'safari'
+
+        message_ids_by_model = defaultdict(list)
```

Comment:
This is not `message_ids_by_model` but `record_ids_by_record_name`. So in other words, they should be recordset instead of list.